### PR TITLE
Collapse Component

### PR DESCRIPTION
**Pull request by asaloff**
## Description
This PR adds the Collapse component. It also adds the collapse to the existing docs page's code examples.

### CODE REVIEW

File: src/components/collapse/Collapse.svelte [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+  // on:change event
+  $: dispatch("change", {
+    currentActiveKey: $collapseStore.activeKey
+  });
+
+  $: dispatch("click", {
+    lastKeyClicked: $lastKeyClickedStore.key
+  });
+
+  $: iconPositonClassName = (function() {
```
Comment (kujtimiihoxha) [inline src/components/collapse/Collapse.svelte:73]: Can we make this a normal function and just call it here?

File: src/components/icons/base/Icon.svelte [hunk 2]
```diff
@@ -89,6 +89,8 @@
       "ms-transform": `rotate(${rotate}deg)`,
       transform: `rotate(${rotate}deg)`
     };
+  } else {
```
Comment (kujtimiihoxha) [inline src/components/icons/base/Icon.svelte:92]: Haaa good catch, this was not reactive right?

File: src/components/collapse/Collapse.svelte [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+    activeKey:
+      activeKey ||
+      (typeof defaultActiveKey === "object"
+        ? defaultActiveKey
+        : [defaultActiveKey]),
+    isAccordion: !!accordion,
+    expandIcon,
+    usingCustomActiveKey: activeKey
+  });
+  setContext("collapseStore", collapseStore);
```
Comment (kujtimiihoxha) [inline src/components/collapse/Collapse.svelte:58]: can we move this above `$: collapseStore.set` (this is just me being picky 😂)

**Comment by kujtimiihoxha** [comment]
Also Copy to Clipboard is not working for me on examples.

File: src/components/collapse/CollapsePanel.svelte [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+  // Base classes
+  let baseClasses;
+  // Content classes
+  let contentClasses;
+
+  const config = getContext(CONFIG_KEY) || configProvider();
+  const { getPrefixCls } = $config;
+  const prefixCls = getPrefixCls("collapse");
+
+  $: active = $collapseStore.activeKey.includes(key);
```
[the remark this answers is not in the document]
Comment (asaloff) [reply src/components/collapse/CollapsePanel.svelte:58]: Looks like the `$:` is required. I removed them from all those properties and I'm getting `activeIcon is not defined` when I do `activeIcon = $collapseStore.acitveIcon`.

File: src/components/collapse/CollapsePanel.svelte [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+    if (active) {
+      collapseStore.set({
+        ...$collapseStore,
+        activeKey: usingCustomActiveKey
+          ? $collapseStore.activeKey
+          : $collapseStore.activeKey.filter(storedKey => storedKey !== key)
+      });
+    } else {
+      if (accordion) {
+        collapseStore.set({
```
[the remark this answers is not in the document]
Comment (asaloff) [reply src/components/collapse/CollapsePanel.svelte:90]: Yup that seems to work. Good to know!

File: src/components/collapse/CollapsePanel.svelte (hunk 4 above)
[the remark this answers is not in the document]
Comment (kujtimiihoxha) [reply src/components/collapse/CollapsePanel.svelte:58]: You need to define it somewhere at the top with let.

File: src/components/icons/base/Icon.svelte (hunk 2 above)
Comment (asaloff) [reply src/components/icons/base/Icon.svelte:92]: Yes I believe so. This way it'll rotate back if it's set to `null`

File: src/components/collapse/Collapse.svelte (hunk 3 above)
Comment (asaloff) [reply src/components/collapse/Collapse.svelte:58]: 😆 This makes sense though! I'll move it 💯

**Pull request merged**